index.js

```diff
@@ -59,13 +60,18 @@ module.exports = {
   },
 
   _getComparisonPlugin() {
-    const trueEmberVersion = this.emberVersion.match(/\d+\.\d+\.\d+/)[0];
+    const trueEmberVersion = extractTrueVersion(this.emberVersion);
 
-    return [require.resolve('./comparision-plugin.js'), { emberVersion: trueEmberVersion }];
+    return [require.resolve('./comparision-plugin.js'), { emberVersion: trueEmberVersion, root: this.project.root }];
   },
 
   _getDebugPlugin(emberVersion, parentChecker) {
-    const trueEmberVersion = emberVersion.match(/\d+\.\d+\.\d+/)[0];
+    const trueEmberVersion = extractTrueVersion(emberVersion);
+    const emberDataVersion = parentChecker.for('ember-data', 'npm').version;
+    const MODEL_DATA_RELEASES_FLAG_OFF = ['3.3.0-beta.1', '3.4.0-beta.1', '3.4.0-beta.2', '3.4.0-beta.3', '3.4.0-beta.4', '3.4.0', '3.4.1', '3.4.2'];
+    const MODEL_DATA_RELEASES_FLAG_ON = ['3.5.0-beta.1', '3.5.0-beta.2', '3.5.0-beta.3'];
+
+    const ALL_MODEL_DATA_RELEASES = [].concat(MODEL_DATA_RELEASES_FLAG_OFF, MODEL_DATA_RELEASES_FLAG_ON);
```

Comment:
Why do we need three different values (flag off, flag on, and combined)?

Reply:
I wanted to make it clear which ones would *also* require the end user to turn on the flag, ideally we could check the state of the flag but that seemed too messy.

Reply:
I don't think we should do anything before 3.5.0, anything else is a crap shoot of compatibility nightmares for addons trying to be compatible...